torch/distributed/distributed_c10d.py

```diff
@@ -1759,15 +1765,25 @@ def broadcast_object_list(object_list, src=0, group=None):
     else:
         object_sizes_tensor = torch.empty(len(object_list), dtype=torch.long)
 
+    # Current device selection.
+    # To preserve backwards compatibility, ``dist_device`` is default to ``None``
+    # in which case we run current logic of device selection, i.e.
+    # ``current_device`` is CUDA if backend is NCCL otherwise CPU device. In the
+    # case it is not ``None``e we move all intermediate tensors to this given
+    # device. See https://github.com/pytorch/pytorch/issues/60062
     group_backend = get_backend(group)
     is_nccl_backend = group_backend == Backend.NCCL
-    current_device = torch.device("cpu")
+    current_device = None
+    if dist_device is not None:
+        current_device = dist_device
```

Comment:
Given the discussion [here](https://github.com/pytorch/pytorch/pull/61305#discussion_r666599220), since NCCL backend requires the device to be a CUDA device, I think it would be better to check here that `dist_device` is a CUDA device (e.g. `dist_device.type == "cuda"`) if `is_nccl_backend` and explicitly raise an error (e.g. `ValueError`) if not to avoid erroring silently.

Reply:
Got it, added a check